### PR TITLE
Devtooling-162: Refactored idp salesforce

### PR DESCRIPTION
**Pull request by dginty4**

### CODE REVIEW

File: genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+type idpSalesforceProxy struct {
+	clientConfig             *platformclientv2.Configuration
+	identityProviderApi      *platformclientv2.IdentityProviderApi
+	getIdpSalesforceByIdAttr getIdpSalesforceByIdFunc
+	updateIdpSalesforceAttr  updateIdpSalesforceFunc
+	deleteIdpSalesforceAttr  deleteIdpSalesforceFunc
+}
+
+// newIdpSalesforceProxy initializes the idp salesforce proxy with all of the data needed to communicate with Genesys Cloud
+func newIdpSalesforceProxy(clientConfig *platformclientv2.Configuration) *idpSalesforceProxy {
```
Comment (carnellj-genesys) [inline genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go:32]: No getAll functionality

**Review by carnellj-genesys — APPROVED** [review]

File: genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go (hunk 1 above)
Comment (dginty4) [reply genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go:32]: My understanding of this resource is that there is no get all. If a salesforce object exists there is only one of them so the exporter just checks if that object exists or not

File: genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+func getIdpSalesforceProxy(clientConfig *platformclientv2.Configuration) *idpSalesforceProxy {
+	if internalProxy == nil {
+		internalProxy = newIdpSalesforceProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getIdpSalesforceById returns a single Genesys Cloud idp salesforce by Id
+func (p *idpSalesforceProxy) getIdpSalesforceById(ctx context.Context) (idpSalesforce *platformclientv2.Salesforce, resp *platformclientv2.APIResponse, err error) {
```
Comment (charliecon) [inline genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go:54]: "ById" isn't really appropriate here. `getIdpSalesforce` makes more sense

**Review by charliecon — APPROVED** [review]

File: genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go (hunk 2 above)
Comment (dginty4) [reply genesyscloud/idp_salesforce/genesyscloud_idp_salesforce_proxy.go:54]: Thanks Charlie, I forgot to change that

**Pull request merged**